dix/req.c

```diff
@@ -1625,15 +1625,42 @@ static void dix_rop_completed(struct m0_sm_group *grp, struct m0_sm_ast *ast)
 	struct m0_dix_rop_ctx *rop_del_phase2 = NULL;
 	bool                   del_phase2 = false;
 	struct m0_dix_cas_rop *cas_rop;
+	int		       cas_failures = 0;
+	int		       cas_success = 0;
+	int 		       ret_val;
+
+	M0_ENTRY();
 
 	(void)grp;
 	if (req->dr_type == DIX_NEXT)
 		m0_dix_next_result_prepare(req);
-	else
+	else {
+		M0_LOG(M0_DEBUG, "Total CAS requests: %"PRIu64, rop->dg_cas_reqs_nr);
+		if (rop->dg_cas_reqs_nr > 1) {
+			m0_tl_for(cas_rop, &rop->dg_cas_reqs, cas_rop) {
+				ret_val = cas_rop->crp_creq.ccr_sm.sm_rc;
+				if (ret_val != 0) {
+					cas_failures++;
+			 	} else
+					cas_success++;			
+			} m0_tl_endfor;
+		}
+		M0_LOG(M0_DEBUG, "CAS successes: %d, CAS failures: %d", 
+		       cas_success, cas_failures);
```

Comment:
This patch allows DIX to send CAS requests to CAS services that experienced transient/permanent failure. The original code would skip sending units in this case.

Please update "Operation in degraded mode" section of dix/client.h accordingly.

Reply:
@ivan-alekhin 
No. The original code is marking as dirty  if one node is down. Due to this reason unable to set dix pool version while bucket create and operation is failed.  
That means DIX data will not be available if one node is down in N nodes cluster (1+2+0). Here we are not supporting redundancy/protection policy.  Could you check comments in 23915 issue?

I am not sure,  How could marked pool version as dirty if one node is down?  What's the use of redundancy/protection policy? 

If DIX is configured as a 1+2+0(N+K+S), means we have to support DIX requests with out any error if one node health status is good. 

>>Please update "Operation in degraded mode" section of dix/client.h accordingly.
Will update it.